### ISSUE TITLE
Define workflow for Plan Edit locking

### ISSUE THREAD

**Issue by vyruss** [DMPRoadmap/roadmap#105]

**Comment by briri** [comment]
Going to meet with @stephaniesimms and @JEK-III tomorrow to discuss. 

@vyruss and @raycarrick-ed, am I correct in assuming that a conflict would be detected after form submission?

**Comment by raycarrick-ed** [comment]
The flow would be something like:

user edits plan and fills in answer, clicks save, this goes to answers#update which creates a new or updated answer object and calls save and at that point the exception gets thrown. So you're catching it in answers#update

The current answers controller just does  a redirect back to the edit page whether the save works or not (redirect_to :back) but adds in an error message if it fails. So for now, I'm following that model but just adding in the conflicting text and user as well which will do for testing purposes.

**Comment by sjDCC** [comment]
Ray's approach sounds good to me. I assume if it fails the user sees the new text (saved by another beforehand) and can then add their changes? 

Will the notes / comments be handled in the same save function as these may need to be considered at this point too?

**Comment by vyruss** [comment]
@sjdcc comments won't have this issue as you can't edit the same comment as someone else. Please see [this diagram](https://drive.google.com/file/d/0B9CCbVGZCQD3VDY3QUZPV1NtbWM/view) we discussed yesterday on the weekly call.

**Comment by briri** [comment]
We will need to reassess once the substance editor is in place for this one. 

Adding a new comment to a selected section of an answer will add some html markup to the answer so that we can tie the contextualized comment to it. So we may need to have different error message if the user added a comment but someone else had changed the answer while they were commenting.

**Comment by JEK-III** [comment]
Am I right in assuming that the "conflicting text" is the entire answer, not the product of any kind of diff?
Either way–given technical limitations and the infrequency with which situation is expected to arise– this approach sounds good.

**Comment by vyruss** [comment]
You are right @JEK-III, we are not expecting (or require) our users to be familiar with the concept of a diff.
